pkgs/by-name/up/upsun/package.nix

```diff
@@ -0,0 +1,36 @@
+{ stdenvNoCC, lib, fetchurl }:
+
+let
+  os = "linux";
+  arch = if stdenvNoCC.isAarch64 then "arm64" else "amd64";
+  platform = "${os}_${arch}";
+in
+stdenvNoCC.mkDerivation rec {
+  pname = "upsun";
+  version = "5.0.12";
+
+  src = fetchurl {
```

Comment:
```nix
src = {
  x86_64-linux = fetchurl {
    url = "https://github.com/platformsh/cli/releases/download/${version}/upsun_${version}_linux_amd64.tar.gz";
    hash = "sha256-svEPMVY7r7pAoXwFIMYqCEduqR3Nkocaguf2nIGt+G8=";
  };
  aarch64-linux = fetchurl {
    url = "https://github.com/platformsh/cli/releases/download/${version}/upsun_${version}_linux_arm64.tar.gz";
    hash = "sha256-ZraS/PqSPL/kcj5o6hzDdL70IV2IWXOma6OHCiXIDQc=";
  };
}.${stdenvNoCC.system} or (throw "${pname}-${version}: ${stdenvNoCC.system} is unsupported.");
```

This would be cleaner. Also `sha256` is deprecated, use `hash` with SRI hash instead.